src/GraphQL/Validation/Errors/NoUnusedFragmentsError.cs

```diff
@@ -12,12 +13,12 @@ public class NoUnusedFragmentsError : ValidationError
         /// <summary>
         /// Initializes a new instance with the specified properties.
         /// </summary>
-        public NoUnusedFragmentsError(ValidationContext context, FragmentDefinition node)
-            : base(context.Document.OriginalQuery!, NUMBER, UnusedFragMessage(node.Name), node)
+        public NoUnusedFragmentsError(ValidationContext context, GraphQLFragmentDefinition node)
+            : base(context.OriginalQuery!, NUMBER, UnusedFragMessage(node.FragmentName.Name), node)
         {
         }
 
-        internal static string UnusedFragMessage(string fragName)
```

Comment:
@Shane32 Actually I can revert all string->ROM changes in error messages and use `UnusedFragMessage(node.FragmentName.Name.StringValue)`. `ToString()` method is still called when formatting a string, so it is more profitable to use a cached value from `StringValue` (if exist). Although it is not so important when an exception occurs. What do you think?

Reply:
I think:
- ExecutionResult should not contain any ROMs (which would contain references to the entire original query string).
- Parser should not perform caching of any value, string or otherwise -- or else it should always cache and return the string and not hold a reference to the ROM.
- Boxing a ROM should almost always be considered worse than creating a string
- Any speed/memory issues with errors or Apollo Tracing is inconsequential (within reason)

Reply:
I understand and agree with the last point. For others the question is - why?

Reply:
I would ask the opposite - why?

1. Unnecessary, memory, complexity. Now if you can get the field names to write as ROMs, which is likely possible (but not complete), then there is at least an argument for it.  Even this is only possible on STJ and only when they are not using a `JsonNamingPolicy`. Otherwise there is no purpose.  It just adds unnecessary complexity to the result.  Plus it prevents the GC from cleaning up the original query.  Every time something is level-2 allocated, it needs to be copied to the level 2 heap, and then is a slower and longer process to release.  Using ROMs everywhere does not help at all.  100 level-1 allocations might be better than 1 large level-2 allocation.
2. Wastes memory with no benefit
3. Wastes memory with no benefit.

Reply:
If the use of ROMs are too deeply embedded in the execution result to extract, and for good reason, then okay, but I still don't think people should need to encounter a ROM during their use of the code.  And now it seems they will, if they want to inspect / change the results at all.

Reply:
1. Regarding original query and cleaning up the original query. I have some plans to switch from `string` to to memory buffers for it. You can view this PR as a first step towards this. I can not say anything defined yet though I understand your considerations.
2.
> so either (a) no allocations take place, or (b) allocations are required

I would say that my goal is to minimize allocations at least when executing a request without errors. And in most cases this goal succeeds. Depending on the size of the strings, the results can be varied.

3. ROMs are boxed in most cases only on error paths. I would not argue about specific numbers.

> And now it seems they will, if they want to inspect / change the results at all.

Of course they will but it is hardly necessary for the majority of consumers. See #2237 . If we want to move in the direction of increasing performance, we will inevitably come to a some streaming model with minimal conversion of one objects to others and probably "direct" writing of results. I consider this PR as a separate step in the chain of the desired design changes on this path.

Reply:
> Ok to leave ROM in ExecutionResult

Closing as resolved